api/v1alpha1/gateway_types.go

```diff
@@ -165,30 +170,44 @@ type ListenerAddress struct {
 // selectors are associated with the Gateway.
 type RouteBindingSelector struct {
 	// NamespaceSelector specifies a set of namespace labels used for selecting
-	// routes to associate with the Gateway. If NamespaceSelector is defined,
-	// all routes in namespaces matching the NamespaceSelector are associated
-	// to the Gateway.
+	// routes to associate with the Gateway.
 	//
 	// An empty NamespaceSelector (default) indicates that routes from any
-	// namespace will be associated to this Gateway. This field is intentionally
-	// not a pointer because the nil behavior (no namespaces) is undesirable here.
+	// Namespace can be selected by this Gateway.
+	//
+	// The OnlyLocalRoutes field takes precedence over this field. This selector
+	// will only take effect when OnlyLocalRoutes is false.
 	//
 	// Support: Core
 	//
+	// +optional
 	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector" protobuf:"bytes,1,opt,name=namespaceSelector"`
+	// OnlyLocalRoutes is a boolean used to indicate if routes should only be
+	// selected from within the same namespace.
+	//
+	// When true (default) only routes in the same namespace as the Gateway
+	// will be selected.
+	//
+	// This field takes precedence over the NamespaceSelector field. That
+	// selector should only take effect when this field is set to false.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:default=true
+	// +optional
```

Comment:
@robscott I believe this field should be required when `+kubebuilder:default=true` is set.

Reply:
I thought that too but unfortunately that doesn't seem to be possible with CRDs. When I tried to create a Gateway with that CRD config I got an error saying the field was required and the defaulting was not applied. I should create an upstream issue because the optional + default does seem less than ideal.

Reply:
Good to know, thanks!